### DISCUSSION TITLE
Create3 ROS 2 Humble topics not listing on Raspberry Pi 4 Ubuntu server 22.04

### DISCUSSION THREAD

**Discussion by jomoengineer** [iRobotEducation/create3_docs#529] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Raspberry PI 4

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.4

### Which RMW is your robot running?

FastRTPS

### Does your robot have an assigned namespace? If so, please share.

no

### Is the robot connected to a network? If so, what is the network type?

Yes. Via Wifi (wlan0)

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

I have performed a fresh install of Ubuntu Server 22.04 on a Raspberry Pi 4 and have gone through the ROS 2 Humble config and added it to my Create. However, I am not seeing the Create3 topics show when listed.
I can ping the Robot IP 192.168.186.2 from the RasPi 4, but still no go.

Am I missing a step somewhere?

I have tried to run the run the troubleshooting.py script, but it calls iwgetid which is not installed, so I commented this and just added a generic wifi values since this should not be necessary. 

This is what I see when I run 'ros2 topic list'

> $ ros2 topic list
> /parameter_events
> /rosout
> 

This is the output of the troubleshooting.py script:

> $ python3 troubleshooting.py 
> This program will cross-check all relevant settings on your computer and robot to expedite troubleshooting. Please fully reboot the robot and wait for the start up chime before proceeding
> Have you rebooted and waited for the chime? (Y/n) y
> Enter IP address of Robot: 192.168.186.2
> 
>  Testing Computer Configurations
>     Computer Linux Distro: Ubuntu 22.04.3 LTS
>     Computer ROS Distro: Humble
>     Computer RMW: rmw_fastrtps_cpp
>     Computer Wi-Fi Network: SomeWifi
>     Computer Domain ID: 0
> 
>  Testing Robot Configurations
>     Robot ROS Distro: Humble
>     Robot Firmware: H.2.4
>     Robot RMW: rmw_fastrtps_cpp
>     Robot Wi-Fi Network: Not found in logs
>     Robot Domain ID: 0
>     Robot Namespace: None
>     Discovery Server: Unknown
> 
> Summary of Troubleshooting Tests 
> 
> ROS 2 and Linux Distribution Compatibility OK
> 
> ROS 2 Distribution Comparison: OK
> 
> ROS 2 Middleware (RMW) Comparison: OK
> 
> ROS 2 Domain ID Comparison: OK
> 
> Wi-Fi Connection:
> WARNING! No Wi-Fi network was found in the robot's logs. This means one of two things:
>             1. The robot is not connected to Wi-Fi. If you wish to use the robot on Wi-Fi go to the 
>             connect page on the robot's webserver and connect to a network.
>             2. The logs are too old to show the most recent connection. If you believe the robot
>             is connected to a Wi-Fi network please fully reboot the robot, wait for the 
>             startup chime, and run this script again.
>

**Comment by jomoengineer** [comment, marked as answer]
Okay, I solved the issue.

I ended up running

> $ rosdep init
> $ rodep update

And now I can see the topics:

> $ ros2 topic list
> /battery_state
> /cliff_intensity
> /cmd_audio
> /cmd_lightring
> /cmd_vel
> /dock_status
> /hazard_detection
> /imu
> /interface_buttons
> /ir_intensity
> /ir_opcode
> /kidnap_status
> /mobility_monitor/transition_event
> /mouse
> /odom
> /parameter_events
> /robot_state/transition_event
> /rosout
> /slip_status
> /static_transform/transition_event
> /stop_status
> /tf
> /tf_static
> /wheel_status
> /wheel_ticks
> /wheel_vels
>

**Reply by LuukLord** [reply]
thanks this worked. had the same problem.

**Comment by vmtb** [comment]
Save issue 😀

**Comment by fteoharevic** [comment]
I'm really struggling with the same issue and this didn't work for me. Could you possibly go through the steps you take from fresh install to running ros2 topic list? @jomoengineer
